plugins/redfish/meson.build

```diff
@@ -1,7 +1,6 @@
-if get_option('plugin_redfish')
-if not get_option('curl')
-  error('curl is required for plugin_redfish')
-endif
+libcurl_redfish = dependency('libcurl', version : '>= 7.56.0', required: get_option('plugin_redfish'))
```

Comment:
This slightly changes semantics, because before you considered it an error if regular curl support was disabled, but now they are mutually independent options.

That is probably fine assuming you're okay with it. :D The fact that those options used to be bound together can be considered an unfortunate implementation detail, I guess.

I see in a couple other places, you also disentangled options that used to be cascading together but now allow mutually independent behavior.

Reply:
> That is probably fine assuming you're okay with it. :D The fact that those options used to be bound together can be considered an unfortunate implementation detail, I guess.

Yup, this was a semantics change I noticed when putting this together too.  I am fine with it.  It's very unlikely that someone would want to disable the daemon curl support but enable the plugin curl support, but yeah now that will work.

> I see in a couple other places, you also disentangled options that used to be cascading together but now allow mutually independent behavior.

Yeah the other really big one I need to fix is gudev.  A bunch of plugins don't "truly" have a compile dependency on it, but if the daemon doesn't have it they won't work.  Probably the last thing I will attack in this PR :)